decompiler/backend/codevisitor.py

```diff
@@ -70,10 +71,30 @@ def visit_condition_node(self, node: ast_nodes.ConditionNode) -> str:
         true_str = self.visit(node.true_branch_child)
         if node.false_branch is None:
             return f"if ({self._condition_string(node.condition)}) {{{true_str}}}"
+
         false_str = self.visit(node.false_branch_child)
-        if isinstance(node.false_branch_child, ast_nodes.ConditionNode):
-            return f"if ({self._condition_string(node.condition)}){{{true_str}}} else {false_str}"
-        return f"if ({self._condition_string(node.condition)}){{{true_str}}} else{{{false_str}}}"
+
+        condition = node.condition
+        true_child = node.true_branch_child
+        false_child = node.false_branch_child
+
+        def swap_branches():
+            nonlocal condition, true_child, false_child, true_str, false_str
+            condition = ~condition
+            true_child, false_child = false_child, true_child
+            true_str, false_str = false_str, true_str
```

Comment:
I see why you prefer it this way. But perhaps a dataclass containing all this information that also can do the swap_branches would be a cleaner option, i.e. 
```python
@dataclass
class ConditionNodeHelper # not a perfect name
    condition
    true_child
    false_child
    true_string
    false_string

    def swap_branches(self){
        self.condition = ~self.condition # this is quite expensive if we would do this two times. But nor sure whether we could/should optimize this
        self.true_child, self.false_child = ...
        self.true_str, self.false_str = ...
    }
```

Reply:
I've changed the code again, to work a little differently, which removed the need for a helper class and doesn't result in unnecessary negations of the condition.